### PR TITLE
refactor: fix OpenAPI schemas and improvements

### PR DESCRIPTION
**Pull request by ByteBaker**
### **User description**
- **OpenAPI schema fixes**: Replace generic `Object` content types with proper typed schemas
    - `ai/prompt.rs`: Use `UpdatePromptRequest` instead of `Object`
    - `authz/fga.rs`: Use `RoleRequest` for role operations
    - `keys/mod.rs`: Use `KeyAddRequest` for cipher key operations
    - `dashboards/mod.rs`: Use versioned dashboard schemas for create/update endpoints
    - `actions/action.rs`: Remove incorrect request_body specifications

  - **Deprecated endpoint disambiguation**: Add "Deprecated" suffix to alert operation IDs to prevent OpenAPI conflicts

- **Dashboard macro refactoring**: Replace repetitive getter/setter methods with declarative macros
    - `make_getter!` macro generates type-safe field accessors across all dashboard versions (v1-v5)
    - `make_setter!` macro generates unified field mutators with version-specific matching
    - Add comprehensive unit tests covering edge cases (empty values, unsupported versions, missing inner objects)

  - **Pipeline schema**: Add `ToSchema` derive to Pipeline struct for API documentation

  - **Enterprise feature gating**: Properly gate cipher key endpoints with conditional compilation

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Replace OpenAPI Object with typed schemas

- Introduce dashboard getter/setter macros

- Gate cipher key endpoints by feature

- Add ToSchema for Pipeline and multipart

___

### Diagram Walkthrough

```mermaid
flowchart LR
  OpenAPI["OpenAPI annotations"] -- "use typed schemas" --> Handlers["HTTP handlers"]
  DashModel["Dashboard struct"] -- "generate getters/setters" --> Macros["make_getter!/make_setter!"]
  Pipeline["Pipeline model"] -- "derive ToSchema" --> Docs["API Docs"]
  Keys["Cipher key endpoints"] -- "cfg(feature=enterprise)" --> Gated["Enterprise-only"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>lib.rs</strong><dd><code>Enable macro metavariable concat feature</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-c1f72c148113548ea854a05d09e3576632fcd7367279563bf3f42f0aef5454e2">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>mod.rs</strong><dd><code>Add getter/setter macros and unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-5bf716db7e1d91dfbd427a2e320ce407c4f0c92d55d68f8e636daed1b02abd56">+219/-143</a></td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Gate endpoints and use KeyAddRequest parsing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-043222e433d1e5825a4b7bfc2c7feedba417f2630f0082fa67aff64976736396">+137/-138</a></td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td><strong>mod.rs</strong><dd><code>Derive ToSchema for Pipeline</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-c4bd542e4e07ed7706b0643086f5a7d4edbb3647c7ceedbe4bea516c5274d8bd">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>action.rs</strong><dd><code>Remove incorrect request_body annotations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-616df3014c553f763ea57e925f1bff7bb47f9467c632940d93a45b7b1e4c3a07">+0/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>prompt.rs</strong><dd><code>Use UpdatePromptRequest in OpenAPI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-4435a6dbd41255db8c334941643b22ef4eb401004370566cc56394d2e74bab6b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>deprecated.rs</strong><dd><code>Disambiguate deprecated alert operation IDs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-3a5d80eac45bff881716e2cf099f90130659576eca017ad238a61781d3341b7e">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>fga.rs</strong><dd><code>Use RoleRequest schema for roles</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-e7c99f37f3abbb5e0507d2760bbf67df010d9218704b38dce84c5477acecd22e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Switch dashboard request bodies to versioned schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-1b3c257e1d174c826718499faa4a1d4e93465dd56cb2b7b273b690ff7e3223fd">+8/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pipeline.rs</strong><dd><code>Use Pipeline schema in OpenAPI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-c69f4f94eee97ce226caddc497b2adf89e26869f2d72b0d0206dc31ab2714daf">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ingest.rs</strong><dd><code>Add ToSchema and schema for multipart fields</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-40fa0a7da7845799e5bed864e8ccc8111330a3d76fcc8e50af0b76251d566b45">+7/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Replace Object with concrete search request schemas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-96630a0dbb6dd7b2b0248fa325c7c7041ce31340c32826675a4a70772e9d29f6">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>multi_streams.rs</strong><dd><code>Rename operation IDs and use typed requests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-c3233cdf1e68cb2455f91457719b9c2f477e735ee16546fdfe111e76768bd054">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>search_job.rs</strong><dd><code>Rename operation ID and type request body</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-d848b311eae318d198558c29a007c8973afb5f8867a31ec056f4e037dcd1b854">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Remove duplicate org_id param in OpenAPI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-095d70f65b6e69bf7de0180fcbcddcd7c89c4cf00095d7b06f039a07e73916bb">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Use typed request and set operation IDs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-c10dd7b4ea20ffe5840c93a6ae4298aec0c32c8ba1b114192fda6c2bac1c41d1">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>openapi.rs</strong><dd><code>Remove unused CreateDashboardRequestBody schema</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-edb8ce232c0cd08366879f642788a75d99f6bab64b4c0e77391281fdb9241e90">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>distribute_analyze.rs</strong><dd><code>Simplify test assertion with assert!</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-79a2bd6eedd7766e9aaf85c580022139ea164c419a6c86068adcd46a8cb80a99">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>mod.rs</strong><dd><code>Use is_multiple_of in test status selection</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8505/files#diff-68837d2f393b453002ec2d95f6882cfcaf70d9ef6547be2c9a890e8b0fc3c559">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: src/config/src/meta/dashboards/mod.rs [hunk 1]
```diff
@@ -43,157 +43,46 @@ pub struct Dashboard {
...
-            2 => self.v2.as_ref().map(|inner| inner.dashboard_id.as_str()),
-            3 => self.v3.as_ref().map(|inner| inner.dashboard_id.as_str()),
-            4 => self.v4.as_ref().map(|inner| inner.dashboard_id.as_str()),
-            5 => self.v5.as_ref().map(|inner| inner.dashboard_id.as_str()),
-            _ => None,
+/// Generate elegant setter methods for `Dashboard` fields.
+macro_rules! make_setter {
+    ($field:ident; $($ver:literal $vx:ident),+) => {
+        #[doc = concat!("Set the ", stringify!($field), " field for the dashboard.")]
+        pub fn ${concat(set_, $field)}(&mut self, $field: String) {
```
[the remark this answers is not in the document]
Comment (ByteBaker) [reply src/config/src/meta/dashboards/mod.rs:50]: The syntax is valid. It's an experimental feature available on nightly.

File: src/handler/http/request/keys/mod.rs [hunk 2]
```diff
@@ -59,75 +60,73 @@ use crate::common::meta::http::HttpResponse as MetaHttpResponse;
...
+#[cfg(not(feature = "enterprise"))]
+/// Store a key credential in db
+#[post("/{org_id}/cipher_keys")]
+pub async fn save(
+    org_id: web::Path<String>,
+    body: web::Bytes,
+    in_req: HttpRequest,
+) -> Result<HttpResponse, Error> {
+    Ok(MetaHttpResponse::forbidden("not supported"))
+}
```
[the remark this answers is not in the document]
Comment (ByteBaker) [reply src/handler/http/request/keys/mod.rs]: This won't work because `KeyAddRequest` is only available on `enterprise` feature. The code is correct though.

File: src/handler/http/request/keys/mod.rs [hunk 3]
```diff
@@ -328,98 +329,96 @@ pub async fn delete(path: web::Path<(String, String)>) -> Result<HttpResponse, E
...
+#[cfg(not(feature = "enterprise"))]
+/// update the credentials for given key
+#[put("/{org_id}/cipher_keys/{key_name}")]
+pub async fn update(
+    _path: web::Path<(String, String)>,
+    _body: web::Bytes,
+    _in_req: HttpRequest,
+) -> Result<HttpResponse, Error> {
+    Ok(MetaHttpResponse::forbidden("not supported"))
+}
```
[the remark this answers is not in the document]
Comment (ByteBaker) [reply src/handler/http/request/keys/mod.rs]: This won't work because KeyAddRequest is only available on enterprise feature. The code is correct though.

**Comment by ByteBaker** [comment]
@greptileai

**Review by Subhra264 — APPROVED** [review]
